app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -918,47 +917,93 @@ else if(needed == SessionFrame.STATE_COMMAND_ID) {
             //overridePendingTransition(R.anim.push_right_in, R.anim.push_right_out);
         }
     }
-    
+
+    /**
+     * Create a form record and launch form entry activity. If there is an
+     * existing incomplete form that uses the same case, ask the user if they
+     * want to edit or delete that one.
+     *
+     * @param state Needed for FormRecord manipulations
+     */
     private void startFormEntry(AndroidSessionWrapper state) throws SessionUnavailableException{
-        try {
-            //If this is a new record (never saved before), which currently all should be 
-            if(state.getFormRecordId() == -1) {
-                    
-                //If form management isn't enabled we can't have these old forms around anyway
-                if(CommCarePreferences.isIncompleteFormsEnabled()) {
-                    //First, see if we've already started this form before
-                    SessionStateDescriptor existing = state.searchForDuplicates();
-                    
-                    //I'm not proud of the second clause, here. Basically, only ask if we should continue entry if the
-                    //saved state actually involved selecting some data.
-                    if(existing != null && existing.getSessionDescriptor().contains(SessionFrame.STATE_DATUM_VAL)) {
```

Comment:
searchForDuplicates checks for SessionStateDescriptors that match the current one, so if the current one doesn't contain 'STATE_DATUM_VAL', then why even bother running searchForDuplicates, right?

Based on this assumption I've re-ordered the code to only run searchForDuplicates if the current SessionStateDescriptor contains 'STATE_DATUM_VAL'